localstack/utils/aws/resources.py

```diff
@@ -0,0 +1,208 @@
+                env=env,
+                region_name=region_name,
+                client=client,
+            )
+    # deploy the API gateway
+    client.create_deployment(restApiId=api_id, stageName=stage_name)
+    return api
+
+
+def create_api_gateway_integrations(
+    api_id, resource_id, method, integrations=None, env=None, region_name=None, client=None
+):
+    if integrations is None:
+        integrations = []
+    if not client:
+        client = connect_to_service("apigateway", env=env, region_name=region_name)
+    for integration in integrations:
+        req_templates = integration.get("requestTemplates") or {}
+        res_templates = integration.get("responseTemplates") or {}
+        success_code = integration.get("successCode") or "200"
+        client_error_code = integration.get("clientErrorCode") or "400"
+        server_error_code = integration.get("serverErrorCode") or "500"
+        request_parameters = integration.get("requestParameters") or {}
+        # create integration
+        client.put_integration(
+            restApiId=api_id,
+            resourceId=resource_id,
+            httpMethod=method["httpMethod"],
+            integrationHttpMethod=method.get("integrationHttpMethod") or method["httpMethod"],
+            type=integration["type"],
+            uri=integration["uri"],
+            requestTemplates=req_templates,
+            requestParameters=request_parameters,
+        )
+        response_configs = [
+            {"pattern": "^2.*", "code": success_code, "res_templates": res_templates},
+            {"pattern": "^4.*", "code": client_error_code, "res_templates": {}},
+            {"pattern": "^5.*", "code": server_error_code, "res_templates": {}},
+        ]
+        # create response configs
+        for response_config in response_configs:
+            # create integration response
+            client.put_integration_response(
+                restApiId=api_id,
+                resourceId=resource_id,
+                httpMethod=method["httpMethod"],
+                statusCode=response_config["code"],
+                responseTemplates=response_config["res_templates"],
+                selectionPattern=response_config["pattern"],
+            )
+            # create method response
+            client.put_method_response(
+                restApiId=api_id,
+                resourceId=resource_id,
+                httpMethod=method["httpMethod"],
+                statusCode=response_config["code"],
+            )
+
+
+def create_kinesis_stream(stream_name, shards=1, env=None, delete=False):
+    env = get_environment(env)
+    stream = KinesisStream(id=stream_name, num_shards=shards)
+    conn = connect_to_service("kinesis", env=env)
+    stream.connect(conn)
+    if delete:
+        run_safe(lambda: stream.destroy(), print_error=False)
+    stream.create()
+    # Note: Returning the stream without awaiting its creation (via wait_for()) to avoid API call timeouts/retries.
+    return stream
```

Comment:
I feel like all service specific stuff over here should be moved to the respective provider module `.utils`, but not necessarily in this PR. Things in `.resources` and `.queries` as well as `.arns`. Any thoughts?

Reply:
Yeah agreed :+1:  We can tackle these separately. Usually they should go hand-in-hand with some refactoring where they're used anyway since  a lot of them don't make much sense, especially now with multi-region and multi-account.